### ISSUE TITLE
MO2 installation error loading shared library

### ISSUE THREAD

**Issue by dontpokethebear3893** [sonic2kk/steamtinkerlaunch#736]
## System Information
- SteamTinkerLaunch version: 12 -->
- Distribution: SteamOS 3.4 Stable on Steam Deck
- Installation Method: ProtonUp-QT
## Issue Description
When assigning it as the compatibility tool for any game (Tested Morrowind + Oblivion), MO2 fails to install. I read the documentation and tried installing using the command line (after performing a full uninstall and reinstall of SteamTinkerLaunch) and got this output after running "steamtinkerlaunch mo2 install":

"Preparing to install SteamTinkerLaunch on Steam Deck
Downloading latest SteamTinkerLaunch

Dependency 'wget' already installed, nothing to do.
Dependency 'innoextract' already installed, nothing to do.
Dependency 'cabextract' already installed, nothing to do.
Dependency 'yad' already installed, nothing to do.

Finished installing SteamTinkerLaunch ('v12.0')!
innoextract: error while loading shared libraries: libboost_iostreams.so.1.78.0: cannot open shared object file: No such file or directory"

It seems like it's selecting GE-Proton7-48 as the default proton used for running the install. More information in the logs shows this:

"Thu Feb  9 11:13:05 AM EST 2023 ERROR - installMO2 - Version for 'innoextract' is invalid. You need to at least version '1.9'
Thu Feb  9 11:13:05 AM EST 2023 ERROR - installMO2 - Starting /home/deck/.config/steamtinkerlaunch/mo2/compatdata/pfx/drive_c/Modding/MO2/ModOrganizer.exe using wine/proton instead
Thu Feb  9 11:13:05 AM EST 2023 WARN - installMO2 - Unfortunately 'innoextract' is required to install ModOrganizer on the Steam Deck, but it wasn't found
Thu Feb  9 11:13:05 AM EST 2023 WARN - installMO2 - '/home/deck/.config/steamtinkerlaunch/mo2/compatdata/pfx/drive_c/Modding/MO2/ModOrganizer.exe' not found after installation, creating '/dev/shm/steamtinkerlaunch/ModOrganizer-failed.txt' to avoid further attempts"

I'm a little confused as to what I did wrong. It also seems odd that steamtinkerlaunch seems to re-install itself every time I run it from the terminal; I even get the visual pop-ups on the left hand side like I did after initially installing it from ProtonUP-QT. I do not know if that's expected. Let me know if there's anything I can do to get around the issue.

[installMO2_protonrun.log](https://github.com/sonic2kk/steamtinkerlaunch/files/10699251/installMO2_protonrun.log)
[ModOrganizer-failed.txt](https://github.com/sonic2kk/steamtinkerlaunch/files/10699252/ModOrganizer-failed.txt)
[steamtinkerlaunch.log](https://github.com/sonic2kk/steamtinkerlaunch/files/10699253/steamtinkerlaunch.log)

**Comment by frostworx** [comment]
hi,
seems like the Steam Deck OS updated its built-in boost collection to 1.80:

```
(deck@steamdeck ~)$ find /usr/lib/ -name "libboost_iostreams.so.*"
/usr/lib/libboost_iostreams.so.1.80.0
```

your _(and my)_ downloaded/extracted innoextract requires the older 1.78 version though:

```
(deck@steamdeck ~)$ ldd /home/deck/stl/deps/usr/bin/innoextract 
	linux-vdso.so.1 (0x00007ffdaa716000)
	liblzma.so.5 => /usr/lib/liblzma.so.5 (0x00007efe30b48000)
	libboost_iostreams.so.1.78.0 => not found
	libboost_filesystem.so.1.78.0 => not found
	libboost_program_options.so.1.78.0 => not found
	libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007efe30911000)
	libm.so.6 => /usr/lib/libm.so.6 (0x00007efe30827000)
	libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007efe30807000)
	libc.so.6 => /usr/lib/libc.so.6 (0x00007efe30620000)
	/lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007efe30c21000)
```

steamtinkerlaunch _(currently)_ uses 
`https://archive.archlinux.org/packages/i/innoextract/innoextract-1.9-7-x86_64.pkg.tar.zst`
as download source.
but when you download and extract that file and ldd the extracted innoextract binary, you'll see it depends on 1.80,
so I assume our previous download on the steam deck needs to be removed to trigger a fresh update
_(haven't tested this, but should just work)_

a possible workaround would be to add a little consistence check _(for example using ldd)_
in steamtinkerlaunch to detect missing dependencies.

The main problem though is that steamtinkerlaunch is "forced" to download required tools "manually" without using the package management, so this would be an endless issue, which will require manual fixes whenever a dependency breaks.

Iirc I've once seen some 3rd party tool which allows using pacman _(the Arch Linux package management, which is also used by SteamOS, which basically is an Arch Linux)_ for regular users within a separate user environment.
Using that tool would be a more complete solution in theory, but likely it would require all dependencies and the whole base installation in its own environment, so wouldn't help as well.

TL;DR: it should work if you remove 
`/home/deck/stl/deps/usr/bin/innoextract`
and trigger a fresh install which should depend on boost 1.80

~~(you might have to remove the possibly cached download tar.zst file as well, haven't checked or better say I forgot if it ever was cached)~~

edit: doesn't look like a tar.zst is cached, so should be enough to remove the `/home/deck/stl/deps/usr/bin/innoextract`
file

edit 2: using https://github.com/Kron4ek/Conty for all custom programs on the steam deck would be a safe solution as well.
conty support is already implemented in steamtinkerlaunch and therefore "only" needs to be added/enabled for  the steamdeck custom tools.

edit 3: hmh, found the user tool I initally was talking about: https://github.com/fsquillace/junest (never tried it, but might be the perfect tool for the steam deck if it fills the gap properly)

**Comment by sonic2kk** [comment]
To fix this issue, all you should have to do is remove `innoextract` from `~/stl/deps/usr/bin` and force STL to re-download it, which it should do the next time it is launched. This happened whenever SteamOS 3.4 released, as it requires a newer version of innoextract. Existing STL installs did not update. This has been answered before, you can search through closed issues on the GitHub issue tracker in future which should be able to answer your question :-) This was also mentioned on the releases page for SteamTinkerLaunch v12.0, which you should **always read before installing/updating**.

Remove the innoextract dependency and re-run STL, it should re-fetch `innoextract`. Alternatively, removing `~/stl` (or uninstalling from ProtonUp-Qt) and re-installing should also fix the issue.

This issue was already answered in #680 and #716 at least. Searching for `libboost` brings them up, in future it can save you some time to search previously closed issues.

> I'm a little confused as to what I did wrong. It also seems odd that steamtinkerlaunch seems to re-install itself every time I run it from the terminal; I even get the visual pop-ups on the left hand side like I did after initially installing it from ProtonUP-QT. I do not know if that's expected. Let me know if there's anything I can do to get around the issue.

I'm not sure why you would think it's re-installing? SteamTinkerLaunch auto-updates on Steam Deck when not using ProtonUp-Qt, these dialogs are part of SteamTinkerLaunch trying to auto-update but failing as GitHub doesn't correctly bring down the `.git` folder required when ProtonUp-Qt downloads it (GitHub issue).

<hr>

As Frostworx mentions, this is going to be a constant chicken-and-egg issue, which I wanted to solve with #724 but I am not overly happy with the solution I came up with and have not had time to revisit the issue.
